miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/MiniAppSecureStorage.kt

```diff
@@ -0,0 +1,249 @@
+                ).build()
+
+                val inputStream = encryptedFile.openFileInput()
+                val byteArrayOutputStream = ByteArrayOutputStream()
+                var nextByte: Int = inputStream.read()
+                while (nextByte != -1) {
+                    byteArrayOutputStream.write(nextByte)
+                    nextByte = inputStream.read()
+                }
+
+                val plaintext: ByteArray = byteArrayOutputStream.toByteArray()
+                val jsonToRead = plaintext.toString(Charsets.UTF_8)
+                return deSerializeItems(jsonToRead)
+            } else {
+                return null
+            }
+        } catch (e: Exception) {
+            return null
+        }
+    }
+
+    fun deleteSecureStorage(
+        miniAppId: String,
+        onSuccess: () -> Unit,
+        onFailed: (MiniAppStorageError) -> Unit
+    ) {
+        if (isStorageAvailable(miniAppId)) {
+            scope.launch {
+                val file = File(secureStorageBasePath, "$miniAppId.txt")
+                file.delete()
+                if (!file.exists()) {
+                    onSuccess()
+                } else {
+                    onFailed(MiniAppStorageError.failedDeleteError)
+                }
+            }
+        } else {
+            onFailed(MiniAppStorageError.unavailableStorage)
+        }
+    }
+
+    fun deleteSecureStorageItems(
+        miniAppId: String,
+        keySet: Set<String>,
+        onSuccess: (Map<String, String>) -> Unit,
+        onFailed: (MiniAppStorageError) -> Unit
+    ) {
+        scope.launch {
+            storageState.postValue(StorageState.LOCK)
+            if (isStorageAvailable(miniAppId)) {
+                val storedItems = readFromEncryptedFile(miniAppId)
+                storedItems?.let { items ->
+                    val filterItems = items.filter { !keySet.contains(it.key) }
+                    writeToEncryptedFile(
+                        miniAppId,
+                        serializedItems(filterItems),
+                        onSuccess,
+                        onFailed
+                    )
+                } ?: kotlin.run {
+                    onFailed(MiniAppStorageError.emptyStorage)
+                }
+            } else {
+                onFailed(MiniAppStorageError.unavailableStorage)
+            }
+            storageState.postValue(StorageState.UNLOCK)
+        }
+    }
+
+    fun loadSecureStorage(
+        miniAppId: String,
+        onSuccess: (Map<String, String>) -> Unit,
+        onFailed: (MiniAppStorageError) -> Unit
+    ) {
+        scope.launch {
+            storageState.postValue(StorageState.LOCK)
+            if (isStorageAvailable(miniAppId)) {
+                val storedItems = readFromEncryptedFile(miniAppId)
+                storedItems?.let {
+                    onSuccess(storedItems)
+                } ?: kotlin.run {
+                    onFailed(MiniAppStorageError.emptyStorage)
```

Comment:
What is the case where this error will happen? Is it when there is some error loading the file?
I think it should also return `unavailableStorage` in that case.

Reply:
@corycaywood addressed.